crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -192,30 +192,18 @@ impl OnSettlementEventUpdater {
         );
 
         // surplus and fees calculation
-        let domain_separator = self.eth.contracts().settlement_domain_separator();
-        let order_uids = settlement.order_uids(domain_separator)?;
-        let order_fees = order_uids
-            .clone()
-            .into_iter()
-            .zip(Postgres::order_fees(ex, &order_uids).await?)
-            .collect::<Vec<_>>();
-
         let surplus = settlement.total_surplus(&external_prices);
         let (fee, order_executions) = {
-            let all_fees = settlement.all_fees(&external_prices, &order_fees);
-            // total unsubsidized fee used for CIP20 rewards
+            let domain_separator = self.eth.contracts().settlement_domain_separator();
+            let all_fees = settlement.all_fees(&external_prices, domain_separator);
+            // total fee used for CIP20 rewards
             let fee = all_fees
                 .iter()
                 .fold(0.into(), |acc, fees| acc + fees.native);
-            // executed fees for each order execution
+            // executed surplus fees for each order execution
             let order_executions = all_fees
                 .into_iter()
-                .zip(order_fees.iter())
-                .map(|(fee, (_, order_fee))| match order_fee {
-                    // market orders have no surplus fee
-                    Some(_) => (fee.order, 0.into()),
-                    None => (fee.order, fee.sell),
-                })
+                .map(|fee| (fee.order, fee.executed_surplus_fee().unwrap_or(0.into())))
```

Comment:
nit: `order_executions` should probably be renamed to contain that they also contain the surplus fee (and [`ExecutedFee`](https://github.com/cowprotocol/services/blob/29687960e746353ec3af56d59e2135908c39ad9c/crates/autopilot/src/database/on_settlement_event_updater.rs#L11) could probably be renamed into `ExecutedSurplusFee`) for more clarity.

Reply:
I might have looked into the future with this one, where `executed_surplus_fee` would be renamed to `executed_fee` or `executed_network_fee` since it will be the only one that exists, so why having a "surplus" in the name.